lib/data_scripts/update_bangladesh_regions_script.rb

```diff
@@ -0,0 +1,144 @@
+          results[:created][:facility_groups] += 1
+        else
+          logger.warn(errors: facility_group.errors, msg: "Errors trying to save facility group #{facility_group.name}")
+          results[:errors][:facility_groups] += 1
+        end
+      end
+      upazila_region = find_or_create_region(:block, upazila_name, district_region)
+      facility_region = find_or_create_region(:facility, facility_name, upazila_region)
+      facility_attrs = {
+        country: "Bangladesh",
+        business_identifiers: [FacilityBusinessIdentifier.new(identifier_type: :dhis2_org_unit_id, identifier: row[:facilityorganization_code])],
+        district: district_region.name,
+        facility_group: facility_group,
+        facility_size: facility_size,
+        name: facility_name,
+        region: facility_region,
+        state: division_region.name,
+        zone: upazila_region.name
+      }
+      facility = Facility.new(facility_attrs)
+      if run_safely { facility.save }
+        results[:created][:facilities] += 1
+      else
+        logger.warn(errors: facility.errors, msg: "Errors trying to save facility #{facility.name}")
+        results[:errors][:facilities] += 1
+      end
+    end
+
+    results
+  end
+
+  def find_or_create_region(region_type, name, parent)
+    cache[region_type][name] ||= Region.where(region_type: region_type).find_by(name: name) || create_region(region_type, name, parent)
+  end
+
+  def create_region(region_type, name, parent)
+    region = Region.new(name: name, region_type: region_type, reparent_to: parent)
+    if run_safely { region.save }
+      results[:created][:regions] += 1
+    else
+      results[:errors][:regions] += 1
+    end
+    region
+  end
+
+  def destroy_empty_facilities
+    sql = <<-SQL
+      NOT EXISTS (SELECT 1 FROM patients where patients.registration_facility_id = facilities.id) AND
+      NOT EXISTS (SELECT 1 FROM patients where patients.assigned_facility_id = facilities.id)
+    SQL
+    related_facility_groups = []
+    related_users = []
+
+    facilities = Facility.where(facility_size: ["community", nil]).includes(:facility_group, :users).where(sql)
+    logger.info { "Removing #{facilities.size} empty facilities" }
+    facilities.each do |facility|
+      related_facility_groups << facility.facility_group
+      related_users.concat(facility.users)
+      if run_safely { facility.destroy }
+        results[:deleted][:facilities] += 1
+      end
+    end
+    related_users.each do |user|
+      next if user.phone_number_authentications.size > 1
+      if run_safely { user.destroy }
+        results[:deleted][:users] += 1
+      end
+    end
+    logger.info { "Removing facility groups with no facilities" }
+    related_facility_groups.each do |facility_group|
+      if facility_group.facilities.size == 0
+        logger.info { "Removing facility group #{facility_group.name} " }
+        run_safely { facility_group.destroy }
+      end
+    end
```

Comment:
Adding a log here that shows `0` remaining community clinics might be helpful, just in case someone added users/patients to these facilities when we run it.